c_common.c

```diff
@@ -164,11 +167,19 @@ void connect_as_client(const char *ip_addr, const char *port_num, int *sock) {
     serv_addr.sin_addr.s_addr =
         inet_addr(ip_addr);  // the ip_address to connect to
     serv_addr.sin_port = htons(atoi(port_num));
-    if (connect(*sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) ==
-        -1) {
-        error_exit("connect() error!");
+
+    int count_retries = 0;
+    int ret = -1;
+    while (count_retries++ < 100) {
+        ret = connect(*sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+        if (ret < 0) {
+            usleep(500);
```

Comment:
Please add logging for retry.

Reply:
Fixed 7dfd077a15f37d0f8520bc1f1604915023c4e718.